spec(v1.0): Implement Strict Catalog JSON Schema Restrictions and Consolidation (#1629)
google/A2UI: specification/v1_0/docs/a2ui_protocol.md

1. **Strict Top-Level vs. `$defs` Boundary:**
   - **Top-Level components and functions:** All component and function schemas MUST be declared directly under the top-level keys `"components"` and `"functions"` respectively.
   - **External References inside `$defs`:** Any definition referenced externally (e.g., from the envelope schema `server_to_client.json` or `common_types.json`) MUST reside inside the `"$defs"` object at the catalog root. This strictly includes:
     - `surfaceProperties`: Referenced as `catalog.json#/$defs/surfaceProperties`.
     - `anyComponent`: Referenced as `catalog.json#/$defs/anyComponent`.
     - `anyFunction`: Referenced as `catalog.json#/$defs/anyFunction`.
2. **No Custom `$defs` or Helpers:**
   - To prevent unconstrained branching, custom definitions or shared helper schemas inside a catalog are strictly prohibited under `"$defs"`.
   - The only allowed keys within the catalog's `"$defs"` object are `anyComponent`, `anyFunction`, and `surfaceProperties`.

>    - All helper properties (such as the relative layout `weight` property) MUST be inlined directly inside the properties block of each supporting component schema rather than referenced from a shared helper.

Reviewer: ```suggestion
   - All helper properties (such as common properties factored out of catalog items) MUST be inlined directly inside the properties block of each supporting component schema rather than referenced from a shared helper.
```

3. **Restricted `$ref` Targets:**
   - Local `$ref` targets are restricted to referencing the catalog's top-level components or functions (e.g., `#/components/Text`, `#/functions/required`).

>    - External `$ref` targets MUST reference the standard types inside `common_types.json` (`https://a2ui.org/specification/v1_0/common_types.json#/$defs/...`), limited to the following allowed schemas:

Reviewer: Instead of a list that needs to be maintained, couldn't you say:

```suggestion
   - External `$ref` targets MUST only reference the standard types inside `common_types.json` (`https://a2ui.org/specification/v1_0/common_types.json#/$defs/...`), and is limited to that set of schemas.
```

Then we could avoid the list.  I just worry that it will immediately be out of date.

Author: There are a lot of schemas in `common_types.json` which shouldn't be directly referenced from catalogs, e.g. CallId, AccessibilityAttributes (included automatically in Component schema), so I want to be explicit here in the spec. I figure we don't make new specifications very often, so it's okay that we think about this each time.

Reviewer: Hmm.  Okay, I guess that makes sense.  I wonder if we can document that this needs to be revisited when the types change. Somewhere that an agent might notice.

Author: Hmmm yeah! I want to avoid compromising the clarity of this specification by including content that is purely about the higher level task of designing new specifications. But perhaps there can be a `specification-change` skill that notes this.

Filed https://github.com/a2ui-project/a2ui/issues/1689
